cumulus/client/cirrus-executor/src/processor.rs

```diff
@@ -206,17 +210,43 @@ where
 
 		// The applied txs can be fully removed from the transaction pool
 
-		// TODO: win the executor election to broadcast ER, authority node only.
-		let is_elected = self.is_authority;
+		let authority_id = self.primary_chain_client.runtime_api().authority_id(&BlockId::Hash(
+			PBlock::Hash::decode(&mut primary_hash.encode().as_slice())
+				.expect("Primary block hash must be the correct type; qed"),
+		))?;
+
+		if SyncCryptoStore::has_keys(&*self.keystore, &[(authority_id.to_raw_vec(), KEY_TYPE)]) {
+			let to_sign = execution_receipt.hash().encode();
+			match SyncCryptoStore::sign_with(
+				&*self.keystore,
+				KEY_TYPE,
+				&authority_id.into(),
+				&to_sign,
+			) {
+				Ok(Some(_signature)) => {
+					tracing::debug!(
+						target: LOG_TARGET,
+						"Generating the signed execution receipt for #{}",
+						header_hash
+					);
+					// TODO: Wrap an execution receipt with signature and gossip the signed execution receipt over the network.
+
+					if let Err(e) =
+						self.execution_receipt_sender.unbounded_send(execution_receipt.clone())
+					{
+						tracing::error!(target: LOG_TARGET, error = ?e, "Failed to send execution receipt");
+					}
 
-		if is_elected {
-			if let Err(e) = self.execution_receipt_sender.unbounded_send(execution_receipt.clone())
-			{
-				tracing::error!(target: LOG_TARGET, error = ?e, "Failed to send execution receipt");
+					// Return `Some(_)` to broadcast ER to all farmers via unsigned extrinsic.
+					Ok(Some(execution_receipt.into()))
+				},
+				Ok(None) => Err(sp_blockchain::Error::Application(Box::from(
```

Comment:
`Box::new`?

Reply:
`Box::new` can't be used as it requires StdError which is not implemented by String and str according to the compiler.